src/resolvers.ts

```diff
@@ -177,9 +180,17 @@ const resolvers = {
             }
         }) => {
             try {
-                const updatedSpokenLanguages = args.input.spokenLanguages ?
-                    submissionService.mapAndValidateSpokenLanguages(args.input.spokenLanguages)
-                    : undefined
+                let updatedSpokenLanguages: gqlType.SpokenLanguage[] | undefined
+                
+                if (args.input.spokenLanguages) {
+                    const spokenLanguagesResult = submissionService
+                        .mapAndValidateSpokenLanguages(args.input.spokenLanguages)
+                    
+                    if (spokenLanguagesResult.hasErrors) {
+                        throw new Error('Invalid spoken Language provided.')
+                    }
+                    updatedSpokenLanguages = spokenLanguagesResult.data
+                }
```

Comment:
This is ok for now. 
@evan-desu  can you create a new issue to move all of the mapping and validation work into the service? 
The idea is to keep your resolvers as simple as possible and have the service method do all the work.